clients/src/main/java/org/apache/kafka/clients/consumer/internals/FetchRequestManager.java

```diff
@@ -65,16 +67,59 @@ protected void maybeThrowAuthFailure(Node node) {
         networkClientDelegate.maybeThrowAuthFailure(node);
     }
 
+    /**
+     * Request that a fetch request be issued to the cluster to pull down the next batch of records.
+     *
+     * <p/>
+     *
+     * The returned {@link CompletableFuture} is {@link CompletableFuture#complete(Object) completed} when the
+     * fetch request(s) have been created and enqueued into the network client's outgoing send buffer.
```

Comment:
To be accurate, it completes when the fetch request has been generated, not yet enqueued really (because it completes on the finally clause on the manager.poll, and the generated request will be enqueued later on, after all managers are polled), is my understanding correct?

Reply:
Wow, I finally met someone even more pedantic than me 😏

You're correct, I'll update the docs. 😄

Reply:
Updated the comments. Please take a look and let me know if it needs more massaging. Thanks!